shell/platform/windows/windows_lifecycle_manager.cc

```diff
@@ -75,6 +75,24 @@ bool WindowsLifecycleManager::WindowProc(HWND hwnd,
     case WM_DWMCOMPOSITIONCHANGED:
       engine_->OnDwmCompositionChanged();
       break;
+
+    case WM_SIZE:
+      if (wpar == SIZE_MAXIMIZED || wpar == SIZE_RESTORED) {
+        OnWindowStateEvent(hwnd, SHOW);
+      } else if (wpar == SIZE_MINIMIZED) {
+        OnWindowStateEvent(hwnd, HIDE);
+      }
+      break;
+
+    case WM_SHOWWINDOW:
+      if (!wpar) {
+        OnWindowStateEvent(hwnd, HIDE);
+      }
+      break;
+
+    case WM_DESTROY:
+      OnWindowStateEvent(hwnd, HIDE);
+      break;
```

Comment:
Note in case anyone is curious: The top level runner window receives a SETFOCUS message when it first comes up, but then does not receive further SETFOCUS or KILLFOCUS messages when the Flutter application gains or loses focus, so we are not listening for it here.